components/fxa-client/src/http_client.rs

```diff
@@ -3,39 +3,49 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 use crate::{config::Config, error::*};
-use browser_id::{derive_hawk_auth_key_from_session_token, hawk_request::HawkRequestBuilder};
+use hex;
+use rc_crypto::hawk::{Credentials, Key, PayloadHasher, RequestBuilder, SHA256};
+use rc_crypto::{digest, hkdf, hmac};
 use serde_derive::*;
 use serde_json::json;
 use std::collections::HashMap;
+use url::Url;
 use viaduct::{header_names, status_codes, Method, Request, Response};
 
-pub(crate) mod browser_id;
+const HAWK_HKDF_SALT: [u8; 32] = [0b0; 32];
+const HAWK_KEY_LENGTH: usize = 32;
 
 #[cfg_attr(test, mockiato::mockable)]
 pub trait FxAClient {
-    fn oauth_token_with_code(
+    fn oauth_tokens_from_code(
         &self,
         config: &Config,
         code: &str,
         code_verifier: &str,
     ) -> Result<OAuthTokenResponse>;
-    fn oauth_token_with_refresh_token(
+    fn oauth_tokens_from_session_token(
         &self,
         config: &Config,
-        refresh_token: &str,
+        session_token: &str,
         scopes: &[&str],
     ) -> Result<OAuthTokenResponse>;
-    fn duplicate_session(
+    fn oauth_token_with_refresh_token(
         &self,
         config: &Config,
-        session_token: &[u8],
-    ) -> Result<DuplicateTokenResponse>;
-    fn oauth_token_from_session_token(
+        refresh_token: &str,
+        scopes: &[&str],
+    ) -> Result<OAuthTokenResponse>;
+    fn oauth_token_with_session_token(
         &self,
         config: &Config,
-        session_token: &[u8],
+        session_token: &str,
```

Comment:
It felt weird passing other tokens around as strings but this one around as bytes, so I changed it to bytes for simplicity (it's stored in the state as bytes as well).

Reply:
I wonder if we should create some transparent `{Access, Refresh, Session}Token` types to avoid mistakes (we definitely did that mistake once or twice in the paste).